src/index.js

```diff
@@ -7,15 +7,73 @@ import { ThemeProvider } from '@mui/material/styles'
 import { theme } from './components/theme';
 import { Provider } from 'react-redux';
 import store from './store/index';
+import { clusterApiUrl } from '@solana/web3.js';
+import { ConnectionProvider, WalletProvider } from '@solana/wallet-adapter-react';
+
+import { LedgerWalletAdapter, PhantomWalletAdapter, SlopeWalletAdapter, SolflareWalletAdapter, TorusWalletAdapter, UnsafeBurnerWalletAdapter } from '@solana/wallet-adapter-wallets';
+
+import {
+  WalletModalProvider,
+} from '@solana/wallet-adapter-react-ui';
+
+require('@solana/wallet-adapter-react-ui/styles.css');
+
+
 
 const root = ReactDOM.createRoot(document.getElementById('root'))
 
+const SolanaWalletAdapterContext = ({ children }) => {
+  // The network can be set to 'localnet','devnet', 'testnet', or 'mainnet-beta'.
+  const network = process.env.REACT_APP_CLUSTER;
+
+
+  const endpoint = useMemo(() => {
```

Comment:
this method will do the work, if is empty this variable it will get from the provider, right?

Reply:
the purpose of the REACT_APP_CLUSTER variable is basically for setting the url u want the app to connect to on the dapp. once set, it wouldn't be affected by changes on the user wallet. 

meaning even if the user changes their wallet network, it won't be changed on the dapp and the same cluster url that we set will be used regardless of the network 

so we're the ones creating the connection object not the user wallet

Reply:
Yes I get that, however, in payment page, the payment can be generated in any chain(mainnet, polygon, testnet) so we cannot set this is an file, the user must change it in execution time. For instance, if we set this variable as mainnet for solana, anda the user wants to create a test payment in the devnet it won't work. Another example, if you go on uniswap you can swap in the chain you want by changing your wallet connection.